### PR TITLE
test(pie-aperture): DSW-1581 - Add Playwright Tests for Form

### PR DESCRIPTION
**Pull request by jamieomaguire**
This PR adds the following functionality:

- Shared form test thats executed across all applications
- New `playwright:show-report` script to launch Playwright reports after test execution.
- Adds `webServer` configuration to Playwright to automatically execute `yarn dev` before test execution
- Updates README.md to document our test approach for this repo.

### CODE REVIEW

**Review by JoshuaNg2332 — DISMISSED** [review]

File: .github/workflows/ci-nuxtjs.yml [hunk 1]
```diff
@@ -119,13 +119,20 @@ jobs:
...
-  #       run: yarn test:visual --filter=nuxt-app
-  #   env:
-  #     BROWSERSTACK_USERNAME: ${{ secrets.BROWSERSTACK_USERNAME }}
-  #     BROWSERSTACK_ACCESS_KEY: ${{ secrets.BROWSERSTACK_ACCESS_KEY }}
-  #     NUXT_AMPLIFY_ID: d36dan3bxjue8c
-  #     PERCY_TOKEN_PIE_APERTURE_NUXT: ${{ secrets.PERCY_TOKEN_PIE_APERTURE_NUXT }}
-  #     PR_NUMBER: ${{ github.event.number }}
+  #     - shell: bash
+  #     - run: yarn test:visual --filter=nuxt-app
+    env:
```
Comment (xander-marjoram) [inline .github/workflows/ci-nuxtjs.yml:133]: It's not obvious what these apply to, if it's the deploy Nuxt app step is it worth moving them up above the commented out steps?

File: README.md [hunk 2]
```diff
@@ -1,7 +1,61 @@
 # PIE Aperture
 
-External Test apps for PIE Components
+PIE Aperture is designed to be a testbed repo for the [PIE Web component library](https://github.com/justeattakeaway/pie). This repo houses text applications currently for NextJS, Nuxt and vanilla JS. These applications import the web components and ensure that they work as expected when installed in a variety of different frameworks.
```
Comment (xander-marjoram) [inline README.md:3]:
```suggestion
PIE Aperture is designed to be a testbed repo for the [PIE Web component library](https://github.com/justeattakeaway/pie). This repo houses test applications currently for NextJS, Nuxt and vanilla JS. These applications import the web components and ensure that they work as expected when installed in a variety of different frameworks.
```

File: nuxt-app/package.json [hunk 3]
```diff
@@ -5,15 +5,16 @@
     "build": "nuxt build",
     "dev": "nuxt dev",
     "generate": "nuxt generate",
+    "playwright:show-report": "npx playwright show-report ../playwright-reports/nuxt-app-playwright-report",
     "preview": "nuxt preview",
     "postinstall": "nuxt prepare",
-    "test:system": "APP_NAME=nuxt-app npx playwright test -c ../playwright.config.ts --project=system:nuxt",
+    "test:system": "APP_NAME=nuxt-app npx playwright test -c ../playwright.config.ts --project=system",
```
Comment (xander-marjoram) [inline nuxt-app/package.json:11]: Is `--project=system` required if there is only one entry in the projects array?

File: playwright.config.ts [hunk 4]
```diff
@@ -21,33 +23,28 @@ export default {
...
-        grep: /@vanilla/,
-        use: { ...devices['Desktop Chrome'] },
-        testMatch: '**/*.spec.ts',
+        testMatch: `${monorepoRoot}/test/system/**/*.spec.ts`,
     }
-  ],
+],
+
+webServer: process.env.CI ? undefined : {
+  command: `cd ./${process.env.APP_NAME} && yarn dev`,
```
Comment (xander-marjoram) [inline playwright.config.ts:44]: Just checking, have the packages been installed at this point?

File: test/playwright/page-objects/form.page.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import { type Locator, type Page } from '@playwright/test';
+
+export class FormPage {
+    readonly page: Page;
+    readonly usernameField: Locator;
+    readonly emailField: Locator;
+    readonly passworldField: Locator;
```
Comment (xander-marjoram) [inline test/playwright/page-objects/form.page.ts:7]:
```suggestion
    readonly passwordField: Locator;
```

File: test/playwright/page-objects/form.page.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    readonly submitBtn: Locator;
+    readonly outputData: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.usernameField = page.getByTestId('username');
+        this.emailField = page.getByTestId('email');
+        this.passworldField = page.getByTestId('password');
+        this.passwordConfirmationField = page.getByTestId('passwordConfirmation');
+        this.approveSettingsBtn = page.getByTestId('approveSettings').getByTestId('switch-component');
```
Comment (xander-marjoram) [inline test/playwright/page-objects/form.page.ts:21]: Is `Btn` accurate? Looks like it's targeting the switch

File: vanilla-app/form.js [hunk 7]
```diff
@@ -12,8 +12,11 @@ form.addEventListener('submit', (e) => {
     const formData = new FormData(form);
     const data = Object.fromEntries(formData.entries());
 
+    data.approveSettings = formData.has('approveSettings') ? true : false;
```
Comment (xander-marjoram) [inline vanilla-app/form.js:15]: Out of interest, why is the ternary needed?

File: .github/workflows/ci-nuxtjs.yml (hunk 1 above)
Comment (siggerzz) [reply .github/workflows/ci-nuxtjs.yml]: fixed!

**Review by xander-marjoram — APPROVED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

**Pull request merged**